src/core/api/nat64_api.cpp

```diff
@@ -100,6 +100,40 @@ otError otNat64GetCidr(otInstance *aInstance, otIp4Cidr *aCidr)
 }
 #endif // OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE
 
+#if OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE || OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
+otNat64State otNat64GetState(otInstance *aInstance)
+{
+    return static_cast<otNat64State>(
+#if OPENTHREAD_CONFIG_NAT64_BORDER_ROUTING_ENABLE
+        AsCoreType(aInstance).Get<BorderRouter::RoutingManager>().GetNat64PrefixManagerState()
+#else
+        // OPENTHREAD_CONFIG_NAT64_TRANSLATOR_ENABLE is enabled in this case.
+        AsCoreType(aInstance).Get<Nat64::Translator>().GetState()
```

Comment:
How do we guarantee that the state of prefix manager also represent the state of the translator? E.g. enabling translator might fail after enabling prefix manager